followthemoney/statement/entity.py

```diff
@@ -0,0 +1,407 @@
+
+    def pop(self, prop: P, quiet: bool = True) -> List[str]:
+        prop_name = self._prop_name(prop, quiet=quiet)
+        if prop_name is None or prop_name not in self._statements:
+            return []
+        return list({s.value for s in self._statements.pop(prop_name, [])})
+
+    def remove(self, prop: P, value: str, quiet: bool = True) -> None:
+        prop_name = self._prop_name(prop, quiet=quiet)
+        if prop_name is not None and prop_name in self._properties:
+            stmts = {s for s in self._statements[prop_name] if s.value != value}
+            self._statements[prop_name] = stmts
+
+    def itervalues(self) -> Generator[Tuple[Property, str], None, None]:
+        for name, statements in self._statements.items():
+            prop = self.schema.properties[name]
+            for value in set((s.value for s in statements)):
+                yield (prop, value)
+
+    def get_type_values(
+        self, type_: PropertyType, matchable: bool = False
+    ) -> List[str]:
+        combined: Set[str] = set()
+        for stmt in self.get_type_statements(type_, matchable=matchable):
+            combined.add(stmt.value)
+        return list(combined)
+
+    def get_type_statements(
+        self, type_: PropertyType, matchable: bool = False
+    ) -> List[Statement]:
+        combined = []
+        for prop_name, statements in self._statements.items():
+            prop = self.schema.properties[prop_name]
+            if matchable and not prop.matchable:
+                continue
+            if prop.type == type_:
+                for statement in statements:
+                    combined.append(statement)
+        return combined
+
+    @property
+    def properties(self) -> Dict[str, List[str]]:
+        return {p: list({s.value for s in vs}) for p, vs in self._statements.items()}
+
+    def iterprops(self) -> List[Property]:
+        return [self.schema.properties[p] for p in self._statements.keys()]
+
+    def clone(self: SE) -> SE:
+        data = {"schema": self.schema.name, "id": self.id}
+        cloned = type(self)(self.dataset, data)
+        for stmt in self._iter_stmt():
+            cloned.add_statement(stmt)
+        return cloned
+
+    def merge(self: SE, other: EntityProxy) -> SE:
+        try:
+            self.schema = self.schema.model.common_schema(self.schema, other.schema)
+        except InvalidData as e:
+            msg = "Cannot merge entities with id %s: %s"
+            raise InvalidData(msg % (self.id, e))
+
+        if not isinstance(other, StatementEntity):
+            for prop, values in other._properties.items():
+                self.add(prop, values, cleaned=True, quiet=True)
+            return self
+        for stmt in other._iter_stmt():
+            if self.id is not None:
+                stmt.canonical_id = self.id
+            self.add_statement(stmt)
+        self.extra_referents.update(other.extra_referents)
+        return self
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
FTM producers are going to need this, right @pudo? I also have your ftm 'data-lake' spec in mind here too.

```suggestion
    def to_statements_dict(self) -> Dict[str, Any]:
        data: Dict[str, Any] = {
            "id": self.id,
            "schema": self.schema.name,
            "statements": [s.to_dict() for s in self._iter_stmt()],
        }
        return data
```

Reply:
Implemented this with some tests in https://github.com/alephdata/followthemoney/pull/1749/commits/b1a1b4c6e48eb5b8f6294e5c1393829dbcd52c58